### PR TITLE
Detect cgroup mount point at runtime.

### PR DESCRIPTION
**Pull request by kainwen**
In the past we use hard coded path "/sys/fs/cgroup" as cgroup mount
point, this can be wrong when 1) running on old kernels or 2) the
customer has special cgroup mount points.

Now we detect the mount point at runtime by checking /proc/self/mounts.

Signed-off-by: Ning Yu <nyu@pivotal.io>

### CODE REVIEW

File: src/backend/utils/resgroup/resgroup-ops-linux.c [hunk 1]
```diff
@@ -527,6 +535,45 @@ getOvercommitRatio(void)
...
+	{
+		char * p;
+
+		if (strcmp(me->mnt_type, "cgroup"))
+			continue;
+
+		Assert(strlen(me->mnt_dir) < sizeof(cgdir));
+		strncpy(cgdir, me->mnt_dir, sizeof(cgdir));
+
+		p = strrchr(cgdir, '/');
```
Comment (xiong-gang) [inline src/backend/utils/resgroup/resgroup-ops-linux.c]: is it possible me->mnt_dir is not ended with '/'?

File: src/backend/utils/resgroup/resgroup-ops-linux.c [hunk 2]
```diff
@@ -56,8 +60,10 @@ static void writeInt64(Oid group, const char *comp, const char *prop, int64 x);
 static bool checkPermission(Oid group, bool report);
 static void getMemoryInfo(unsigned long *ram, unsigned long *swap);
 static int getOvercommitRatio(void);
+static void detectCgroupMountPoint(void);
 
 static int cpucores = 0;
+static char cgdir[MAX_PATH_LEN];
```
Comment (xiong-gang) [inline src/backend/utils/resgroup/resgroup-ops-linux.c]: should this be inited with all '0'?

Comment (pivotal-ning-yu) [inline src/backend/utils/resgroup/resgroup-ops-linux.c]: static variables are initialized to all 0 by default.

File: src/backend/utils/resgroup/resgroup-ops-linux.c (hunk 1 above)
Comment (pivotal-ning-yu) [inline src/backend/utils/resgroup/resgroup-ops-linux.c]: In theory no, it's guaranteed by the kernel, we can verify it with below case:

```shell
mkdir a b
sudo mount -t tmpfs case-a a
sudo mount -t tmpfs case-b b/
mount | grep case-
```

**Pull request merged**